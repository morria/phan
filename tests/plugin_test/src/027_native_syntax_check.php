<?php
class X{
    public $y = 2;
    public $x = $y;  // Not a *semantically* valid default, but parsable by php-ast
}
