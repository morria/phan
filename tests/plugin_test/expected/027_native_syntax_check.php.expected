src/027_native_syntax_check.php:2 PhanUnreferencedClass Possibly zero references to class \X
src/027_native_syntax_check.php:3 PhanUnreferencedPublicProperty Possibly zero references to public property \X->y
src/027_native_syntax_check.php:4 PhanInvalidConstantExpression Constant expression contains invalid operations
src/027_native_syntax_check.php:4 PhanNativePHPSyntaxCheckPlugin Saw error or notice for php --syntax-check: "Fatal error: Constant expression contains invalid operations"
src/027_native_syntax_check.php:4 PhanUndeclaredVariable Variable $y is undeclared
src/027_native_syntax_check.php:4 PhanUnreferencedPublicProperty Possibly zero references to public property \X->x
